src/messaging/tests/TestReliableMessageProtocol.cpp

```diff
@@ -459,11 +459,6 @@ void CheckUnencryptedMessageReceiveFailure(nlTestSuite * inSuite, void * inConte
     NL_TEST_ASSERT(inSuite, gLoopback.mDroppedMessageCount == 0);
     // Test that the message was dropped by the receiver
     NL_TEST_ASSERT(inSuite, !mockReceiver.IsOnMessageReceivedCalled);
-
-    // Since peer dropped the message, we might have pending acks. Let's clear the table
-    rm->ClearRetransTable(rc);
```

Comment:
But now those retransmissions will be in the table when later tests run and interfere with those tests's results, no?

Reply:
I went the opposite way that uses `kNoAutoRequestAck`, I'm not sure if it matters.

Reply:
Can you please:

1) Document why the kNoAutoRequestAck is there.
2) Add an explicit check that there are no pending retransmits at test end?  `NL_TEST_ASSERT(inSuite, rm->TestGetCountRetransTable() == 0);`